src/teleop_twist_keyboard.py

```diff
@@ -49,61 +52,63 @@
 		'c':(1,.9),
 	      }
 
+
 def getKey():
 	tty.setraw(sys.stdin.fileno())
 	select.select([sys.stdin], [], [], 0)
 	key = sys.stdin.read(1)
 	termios.tcsetattr(sys.stdin, termios.TCSADRAIN, settings)
 	return key
 
-speed = .1
-turn = .2
 
 def vels(speed,turn):
 	return "currently:\tspeed %s\tturn %s " % (speed,turn)
 
 if __name__=="__main__":
     	settings = termios.tcgetattr(sys.stdin)
-	
-	pub = rospy.Publisher('/cmd_vel', Twist)
+
+	pub = rospy.Publisher('cmd_vel', Twist, queue_size = 1)
 	rospy.init_node('teleop_twist_keyboard')
 
+	speed = rospy.get_param("~speed", 0.25)
+	turn = rospy.get_param("~turn", 0.5)
 	x = 0
 	y = 0
 	th = 0
 	status = 0
 
 	try:
-		print msg
-		print vels(speed,turn)
+		print(msg)
+		print(vels(speed,turn))
 		while(1):
 			key = getKey()
-			if key in speedBindings.keys():
+			if key in moveBindings.keys():
+				x = moveBindings[key][0]
+				y = moveBindings[key][1]
+				th = moveBindings[key][2]
+			elif key in speedBindings.keys():
 				speed = speed * speedBindings[key][0]
 				turn = turn * speedBindings[key][1]
 
-				print vels(speed,turn)
+				print(vels(speed,turn))
 				if (status == 14):
-					print msg
+					print(msg)
 				status = (status + 1) % 15
 			else:
-				if key in moveBindings.keys():
-					x = moveBindings[key][0]
-					y = moveBindings[key][1]
-					th = moveBindings[key][2]
-				else:
-					x = 0
-					y = 0
-					th = 0
-					if (key == '\x03'):
-						break
-				twist = Twist()
-				twist.linear.x = x*speed; twist.linear.z = 0; twist.linear.y = y*speed
-				twist.angular.x = 0; twist.angular.y = 0; twist.angular.z = th*turn
-				pub.publish(twist)
-
-	except:
-		print e
+				x = 0
+				y = 0
+				z = 0
```

Comment:
`z` is not being used anywhere

Reply:
It is used in line 106. In case anything goes wrong. You don't want to keep the old speed. But go to zero.

Reply:
It is not being used. Check carefully. `twist.linear.z = 0` instead of `twist.linear.z = z`